geode-core/src/main/java/org/apache/geode/cache/query/internal/ExecutionContext.java

```diff
@@ -647,4 +661,36 @@ public boolean getIsPRQueryNode() {
     return this.isPRQueryNode;
   }
 
+  /**
+   * Check to see if the query execution was canceled. The query gets canceled by the QueryMonitor
+   * if it takes more than the max query execution time or low memory situations
+   */
+  public boolean isCanceled() {
+    return getQueryCanceledException() != null;
+  }
+
+  public CacheRuntimeException getQueryCanceledException() {
+    return canceledException;
+  }
+
+  public void setQueryCanceledException(final CacheRuntimeException queryCanceledException) {
+    this.canceledException = queryCanceledException;
+  }
+
+  /**
+   * This method attempts to reintrepret a {@link QueryExecutionCanceledException} using the
+   * the value returned by {@link #getQueryCanceledException} (set by the {@link QueryMonitor}).
+   *
+   * @throws if {@link #getQueryCanceledException} doesn't return {@code null} then throw that
+   *         {@link CacheRuntimeException}, otherwise throw {@link QueryExecutionCanceledException}
+   */
+  Object reinterpretQueryExecutionCanceledException() {
+    final CacheRuntimeException queryCanceledException = getQueryCanceledException();
```

Comment:
As I said earlier, it seems fishy that this method takes no exception parameter. If it did, then it could probably revert back to being part of the `QueryMonitior` class (since it wouldn't need any object fields at all).

Reply:
This was actually part of default query before these changes, not the query monitor.  it is definitely a bandaid due to inconsistent exception throwing behavior.  Sometimes we throw the canceled exception off the execution context, sometimes we just throw a raw QueryExecutionCanceledException via the gross abomination static throwExceptionIfQueryOnCurrentThreadIsCanceled() method.